docs/flows/flows.puml

```diff
@@ -298,9 +413,124 @@ bitdaoadmin <-- bitdaogovernance: pass/fail vote result
 
 bitdaoadmin -> bitdaogovernance: start execution
 executor <-- bitdaogovernance: execute proposal
-executor -> governanceproxy: set new bitdaogovernance implementation
-executor -> treasuryproxy: set new bitdaotreasury implementation
-executor -> executorproxy: set new executor implementation
+executor -> proxyadmin: call token proxy upgrade to new token implementation
+
+newpage
+
+autonumber "UpgradeSnapshot:[00]"
+
+== Upgrade Snapshot ==
+
+alice -> forum: create snapshot upgrade request forum proposal
+activate forum
+
+guarantor -> forum: get proposals
+guarantor <-- forum: list of proposals
+guarantor -> forum: put positive/negative feedback
+
+community -> forum: get proposals
+community <-- forum: list of proposals
+community -> forum: put positive/negative feedback
+
+alice <-- forum: get positive/negative feedback
+deactivate forum
+
+community --> community: delegate votes
+
+alice -> snapshot: submit snapshot upgrade request proposal
+activate snapshot
+guarantor -> snapshot: get proposals
+guarantor <-- snapshot: list of proposals
+guarantor -> snapshot: vote to release or proportional slash
+
+tokenholders -> snapshot: get proposals
+tokenholders <-- snapshot: list of proposals
+tokenholders -> snapshot: vote on open proposals
+tokenholders <-- snapshot: vote confirmation
+deactivate snapshot
+
+bitdaoadmin -> snapshot: get vote result
+bitdaoadmin <-- snapshot: pass/fail vote result
+bitdaoadmin -> snapshot: upgrade snapshot
+
+newpage
+
+autonumber "UpgradeMultisig:[00]"
+
+== Upgrade Multisig ==
+
+alice -> forum: create multisig upgrade request forum proposal
+activate forum
```

Comment:
suggestion (non-blocking): perhaps for a follow up PR, I suggest that we consider that multiple folks may be changed in the multi-sig. So it may be Alice, Bob, Carol, etc. who all present their pitch for why they should be part of a multi-sig. As an initial naive approach, we can of course assume that multi-sig proceeds a single update at a time (though that impacts security since newcomers can then act against newcomers that are still queued).

question: is this multi-sig upgrade an upgrade of any multi-sig, bitdaoamin multi-sig, treasury, bond, etc.? perhaps a note at the start could suffice to help give context.

suggestion (perhaps blocking): related to the above, if I assume that this multi-sig upgrade assumes that there is already an existing multi-sig, I suggest that this flow start with the _initialization_ of the multi-sig with the admin keys.

Reply:
Agreed, maybe even 5 folks. Add each owner commitment and multisig generation step. Will add init flow in follow-up PR.